analytics/src/date-boundary-test.test.js

```diff
@@ -0,0 +1,20 @@
+test("date year boundaries", () => {
```

Comment:
These tests are a bit strange. We should be testing the previousDay() function, not the js Date class.  Also, for completeness, we should be checking that the other core fields behave as expected around as well. So, for instance, on a leap year 1 Mar -> 29 Feb.  We are testing the day but not the month. Same for all the other relevant fields.

Reply:
done